### PR TITLE
Get basic stats displayed

### PR DESCRIPTION
**Pull request by BirdTho**
Maybe move unverified trees to a pill display next to Verity menuitem
Maybe move user count next to planters menuitem
Maybe move total trees next to trees menuitem

stats display 1

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,7 @@
 .DS_Store
 .vs
 .idea
+.history
```
Comment (BirdTho) [inline .gitignore:16]: To ignore .history directory made by VS code extension

File: client/src/components/common/MenuStats.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    if (!totalTrees) {
+      getTreeCount();
+    }
+
+    if (unprocessedTreeCount === null) {
+      getUnprocessedTreeCount();
+    }
+  }
+
+  shouldComponentUpdate(nextProps) {
```
Comment (BirdTho) [inline client/src/components/common/MenuStats.js]: Only re-render if all values are populated

File: client/src/models/trees.js [hunk 3]
```diff
@@ -84,14 +84,40 @@ const trees = {
         })
       })
     },
+
```
Comment (BirdTho) [inline client/src/models/trees.js:94]: Separation of getting tree count for the sake of not having to fetch trees with it.

File: client/src/models/trees.js [hunk 4]
```diff
@@ -84,14 +84,40 @@ const trees = {
...
     async getTreesAsync(payload, rootState) {
       // Destruct payload and fill in any gaps from rootState.trees
       const { page, rowsPerPage, filter, orderBy, order } = { ...rootState.trees, ...payload }
 
       /*
        * first load the page count
        */
-      
+      if (!rootState.treeCount) {
```
Comment (BirdTho) [inline client/src/models/trees.js:125]: extracted getting tree count from getTreesAsync

File: client/src/models/verity.js [hunk 5]
```diff
@@ -43,13 +43,19 @@ const verity = {
 		/*
 		 * The default value means: image not approved yet, and not rejected yet too
 		 */
-		filter		: new FilterModel({
-			approved		: false,
-			active		: true,
+		filter: new FilterModel({
+			approved: false,
+			active: true,
+    }),
```
Comment (BirdTho) [inline client/src/models/verity.js:49]: actual change is "Unprocessed filter, previous is whitespace fixes.

**Comment by nmcharlton** [comment]
@BirdTho Is this PR ready for review following the latest changes?
If you're still working on it, we can mark the PR as draft for the moment.

**Comment by BirdTho** [comment]
I think it may still be a draft. More revisions needed

**Comment by BirdTho** [comment]
- Display basic stats on homepage
- Auto-update stats when trees are approved/rejected
- Auto-update unverified tree count in Verity page when approving/rejecting trees
- Add redux mechanism in trees and verity models for invalidating tree counts to flag for updates
- Add withData HOC (receives `needsUpdate: boolean` `fetch: function()` and `data: any`, updates on needsUpdate change by calling fetch, passes data and other props to child

**Comment by BirdTho** [comment]
- Update withData to forward refs

**Comment by BirdTho** [comment]
Greenstand admin panel dash

File: client/src/models/verity.js [hunk 6]
```diff
@@ -43,13 +43,27 @@ const verity = {
...
+    verifiedFilter: new FilterModel({
+      approved: true,
+      active: true,
+    }),
+    rejectedFilter: new FilterModel({
+      approved: false,
+      active: false
```
Comment (nmcharlton) [inline client/src/models/verity.js:56]: I believe `active` should be set to `true` here, but will check with @deepwinter

**Review by nmcharlton — COMMENTED** [review]
Superb work!

File: .nvmrc [hunk 7]
```diff
@@ -0,0 +1 @@
+lts/carbon
```
Comment (nmcharlton) [inline .nvmrc]: We can drop this now - we've moved everything in the project up to v12 (erbium) and have `.nvmrc` files under `client` and `server`

File: client/src/models/verity.js (hunk 6 above)
Comment (BirdTho) [reply client/src/models/verity.js:56]: Looking at the api call, when reject tree is called, active is set to false. 
See here: https://github.com/Greenstand/treetracker-admin/blob/a6f7a931e3d3732f71373afaa2f9c4b980829a06/client/src/api/treeTrackerApi.js#L78
Line 88 below it.

File: .nvmrc (hunk 7 above)
Comment (BirdTho) [reply .nvmrc]: Oh, where was this? I didn't have it after a rebase a week ago, new change?

Comment (nmcharlton) [reply .nvmrc]: About an hour ago – it's very cutting edge 😃

Comment (BirdTho) [reply .nvmrc]: I removed my nvmrc, switched back to 12 localy

File: client/src/models/verity.js (hunk 6 above)
Comment (BirdTho) [reply client/src/models/verity.js:56]: Comment, @nmcharlton ?

File: client/src/components/DashStat.container.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    }),
+    dispatch => ({
+      fetch: dispatch.planters.count
+    })
+  ),
+  withData,
+  (El) => ((props) => (<El
+    color={theme.palette.stats.orange}
+    Icon={PeopleOutlineOutlinedIcon}
+    label={'Users'} {...props} />)),
```
[the remark this answers is not in the document]
Comment (BirdTho) [reply client/src/components/DashStat.container.js]: Did!

File: client/src/models/verity.js (hunk 6 above)
Comment (nmcharlton) [reply client/src/models/verity.js:56]: My apologies, you are correct.

**Review by nmcharlton — APPROVED** [review]

**Pull request merged**